src/fides/api/ops/service/privacy_request/request_service.py

```diff
@@ -66,3 +73,52 @@ def cache_data(
                 privacy_request.cache_masking_secret(masking_secret)
     if drp_request_body:
         privacy_request.cache_drp_request_body(drp_request_body)
+
+
+async def poll_server_for_completion(
+    privacy_request_id: str,
+    server_url: str,
+    token: str,
+    *,
+    poll_interval_seconds: int = 30,
+    timeout_seconds: int = 1800,  # 30 minutes
+    client: AsyncClient | None = None,
+) -> PrivacyRequestResponse:
+    """Poll a server for privacy request completion.
+
+    Requests will report complete with if they have a status of canceled, complete,
+    denied, or error. By default the polling will time out if not completed in 30
+    minutes, time can be overridden by setting the timeout_seconds.
+    """
+    url = f"{server_url}{V1_URL_PREFIX}/privacy-request?request_id={privacy_request_id}"
```

Comment:
nit: could use the `PRIVACY_REQUESTS` constant in the `urn_registry` here, just to keep things a bit more dynamic if that route ever changes